devtools/gradle/src/main/java/io/quarkus/gradle/GradleBuildFileFromConnector.java

```diff
@@ -13,33 +14,35 @@
 import org.gradle.tooling.model.eclipse.EclipseExternalDependency;
 import org.gradle.tooling.model.eclipse.EclipseProject;
 
-import io.quarkus.devtools.buildfile.GradleBuildFile;
-import io.quarkus.devtools.writer.ProjectWriter;
+import io.quarkus.devtools.project.buildfile.AbstractGradleBuildFile;
+import io.quarkus.platform.descriptor.QuarkusPlatformDescriptor;
 
-public class GradleBuildFileFromConnector extends GradleBuildFile {
+public class GradleBuildFileFromConnector extends AbstractGradleBuildFile {
 
     private List<Dependency> dependencies = null;
 
-    public GradleBuildFileFromConnector(ProjectWriter writer) {
-        super(writer);
-        // we need to initialize here since there is no other single point of entry
+    public GradleBuildFileFromConnector(final Path projectFolderPath, final QuarkusPlatformDescriptor platformDescriptor) {
+        super(projectFolderPath, platformDescriptor);
+    }
+
+    public GradleBuildFileFromConnector(Path projectFolderPath, QuarkusPlatformDescriptor platformDescriptor,
+            Path rootProjectPath) {
+        super(projectFolderPath, platformDescriptor, rootProjectPath);
     }
 
     @Override
     public List<Dependency> getDependencies() throws IOException {
         if (dependencies == null) {
             EclipseProject eclipseProject = null;
             if (getBuildContent() != null) {
-                if (getWriter().hasFile()) {
-                    try {
-                        ProjectConnection connection = GradleConnector.newConnector()
-                                .forProjectDirectory(getWriter().getProjectFolder())
-                                .connect();
-                        eclipseProject = connection.getModel(EclipseProject.class);
-                    } catch (BuildException e) {
-                        // ignore this error.
-                        e.printStackTrace();
-                    }
+                try {
+                    ProjectConnection connection = GradleConnector.newConnector()
+                            .forProjectDirectory(getProjectFolderPath().toFile())
+                            .connect();
+                    eclipseProject = connection.getModel(EclipseProject.class);
+                } catch (BuildException e) {
+                    // ignore this error.
+                    e.printStackTrace();
```

Comment:
remove if it should be ignored.

Reply:
@maxandersen this is actually existing code, because of too much diff, git won't detect that it's the same file.